pxr/usd/sdf/pathExpression.cpp

```diff
@@ -39,6 +39,11 @@ TF_REGISTRY_FUNCTION(TfType)
     TfType::Define<VtArray<SdfPathExpression>>();
 }
 
+TF_REGISTRY_FUNCTION(VtValue)
+{
+    VtValue::RegisterSimpleCast<std::string, SdfPathExpression>();
+}
+
```

Comment:
We'd prefer to disallow implicit casts from std::string to SdfPathExpression. Construction of an SdfPathExpression may raise a TfError if the input string doesn't parse, so we'd prefer the user be explicit about when they want to do the conversion and be aware of the potential error.

I believe that means we can remove this and the implicit_convertible registration I'll call out below.